CommandLine/XblConfig/Program.cs

```diff
@@ -0,0 +1,861 @@
+                fileList.AddRange(Directory.GetFiles(pathRoot, filename));
+            }
+
+            return fileList;
+        }
+
+        private static string GetETag(string directory, string sandbox)
+        {
+            string filename = Path.Combine(directory, $"{sandbox ?? "Account"}.etag");
+            if (!File.Exists(filename))
+            {
+                return null;
+            }
+
+            using (var file = File.OpenText(filename))
+            {
+                return file.ReadToEnd();
+            }
+        }
+
+        private static string GetETag(IEnumerable<string> files, string sandbox)
+        {
+            foreach (var file in files)
+            {
+                var eTag = GetETag(Path.GetDirectoryName(file), sandbox ?? "Account");
+                if (eTag != null)
+                {
+                    return eTag;
+                }
+            }
+
+            return null;
+        }
+
+        private static void SaveETag(string etag, string directory, string sandbox)
+        {
+            // Save the etag as a hidden file in the directory.
+            string filename = Path.Combine(directory, $"{sandbox ?? "Account"}.etag");
+            if (File.Exists(filename))
+            {
+                File.Delete(filename);
+            }
+
+            using (StreamWriter sw = new StreamWriter(new FileStream(filename, FileMode.Create, FileAccess.Write), Encoding.UTF8))
+            {
+                sw.Write(etag);
+            }
+
+            new FileInfo(filename)
+            {
+                Attributes = FileAttributes.Hidden
+            };
+        }
+
+        private static void PrintValidationInfo(IEnumerable<ValidationInfo> validationList)
+        {
+            var warnings = validationList.Where(c => c.Severity == Severity.Warning);
+            var errors = validationList.Where(c => c.Severity == Severity.Error);
+            if (warnings.Count() > 0)
+            {
+                Console.ForegroundColor = ConsoleColor.Yellow;
+                Console.WriteLine("Warnings:");
+                foreach (var validationInfo in warnings)
+                {
+                    Console.WriteLine($" - {validationInfo.Message}");
+                }
+
+                Console.WriteLine();
+                Console.ResetColor();
+            }
+
+            if (errors.Count() > 0)
+            {
+                Console.ForegroundColor = ConsoleColor.Red;
+                Console.Error.WriteLine("Errors:");
+                foreach (var validationInfo in errors)
+                {
+                    Console.Error.WriteLine($" - {validationInfo.Message}");
+                }
+
+                Console.WriteLine();
+                Console.ResetColor();
+            }
+        }
+
+        #endregion
+
+        #region Model Classes
+
+        private class BaseOptions
+        {
+            [Option("verbose")]
+            public bool Verbose { get; set; }

```

Comment:
>Verbose [](start = 24, length = 7)

is this used anywhere ?

Reply:
Not yet, but it will.

---
In reply to: [175223360](https://github.com/Microsoft/xbox-live-developer-tools/pull/37#discussion_r175223360) [](ancestors = 175223360)